src/orange/detail/UnitInserter.cc

```diff
@@ -187,6 +193,18 @@ SimpleUnitId UnitInserter::operator()(UnitInput const& inp)
         make_builder(&orange_data_->volume_records)
             .insert_back(vol_records.begin(), vol_records.end()));
 
+    // Create BIH tree
+    if (std::all_of(bboxes.begin(), bboxes.end(), [](BoundingBox const& b) {
+            return b;
+        }))
+    {
+        detail::BIHBuilder bih_builder(std::move(bboxes),
+                                       &orange_data_->local_volume_ids,
+                                       &orange_data_->bih_inner_nodes,
+                                       &orange_data_->bih_leaf_nodes);
+        unit.bih_params = bih_builder();
```

Comment:
With the suggested BIH builder change, you'd initialize class data `BIHBuilder build_bih_;` and here call
```suggestion
        unit.bih_params = build_bih_(std::move(bboxes));
```